content/chainguard/chainguard-images/chainguard-registry/pull-through-guides/harbor/index.md

```diff
@@ -0,0 +1,132 @@
+
+If you don't already have one, generate a pull token in your organization:
+
+```shell
+chainctl auth configure-docker --parent <org-name> --pull-token
+```
+
+This returns username and password credentials:
+
+```Output
+To use this pull token in another environment, run this command:
+
+    docker login "cgr.dev" --username "<pull-token-username>" --password "<pull-token-password>"
+```
+
+Note these values down, as you'll need them shortly.
+
+Next, open up the Harbor UI and perform the following steps:
+
+1. Navigate to **Administration > Registries**
+2. Click the **+ NEW ENDPOINT** button..
+3. Configure the endpoint with these details:
+    * **Provider** — There is no specific integration with the Chainguard registry, so here you must select the generic `Docker Registry` provider type.
+    * **Name** — This is used to refer to your repository. You can choose whatever name you like here, but this guide's examples use the name `cgr.dev`.
+    * **Endpoint URL** — This value **must** be `https://cgr.dev`.
+    * **Access ID** — Enter the username value returned by the `chainctl auth` command you just ran.
+    * **Access Secret** — Here, enter the password value returned by the `chainctl auth` command.
+4. Click the **TEST CONNECTION** button to ensure that Harbor can reach `cgr.dev` successfully.
+5. Click **OK** to create the endpoint.
+
+After creating the endpoint, you can move on to creating a proxy cache.
+
+
+## Create a Proxy Cache
+
+A [*proxy* cache](https://goharbor.io/docs/2.1.0/administration/configure-proxy-cache/) allows Harbor to proxy and cache images from the Chainguard registry.
+
+To configure a cache, perform the following steps in the Harbor UI:
+
+1. Navigate to **Projects**.
+2. Click the **+ NEW PROJECT** button.
+3. Create the project with these details:
+    * **Project Name** — This can be whatever you like. This guide uses the name `cgr—proxy`.
+    * **Proxy Cache** — Toggle this on.
+    * **Endpoint** — Choose the endpoint you created in the previous step.
+4. Click **OK**
+
+Following that, you can pull images from the Harbor project like so:
+
+```shell
+docker pull $HARBOR_URL/cgr-proxy/$ORGANIZATION/$IMAGE:$TAG
+```
+
+Be sure to replace the placeholder values (`$HARBOR_URL`, `$ORGANIZATION`, `$IMAGE`, and `$TAG`) to reflect your own setup.
+
+
+## Create a Replication Rule
+
+A [*replication rule*](https://goharbor.io/docs/2.1.0/administration/configuring-replication/create-replication-rules/) is an alternative approach to a proxy cache. This section outlines how to configure a replication rule that copies images from the Chainguard registry to a Harbor project.
+
+First, create a project. From the Harbor UI, navigate to **Projects** and click the **+ NEW PROJECT** button.
+
+Give the project a name (this guide uses the name `cgr-mirror`) and be sure to leave the **Proxy Cache** option switched off.
```

Comment:
Might want to use a list again and remember Access Level and Project quota limits (I guess people might skip over proxy if they don't want it).